sssd_test_framework/utils/tools.py

```diff
@@ -586,6 +586,35 @@ def teardown(self):
 
         super().teardown()
 
+    def wait_for(self, condition: str, body: str = "sleep 1", time: int = 60) -> SSHProcessResult:
```

Comment:
I wouldn't use `sleep 1` as default, empty string will be better. Our use case is mostly to "wait until command succeeds" to avoid potential race condition, but usually it should succeed very quickly. By putting sleep there, we would just delay the test. 

We can even omit the sleep completely, or use something like 5us (`sleep 0.000005`). Omitting it might be better.

Reply:
Rename `time` -> `timeout`

Reply:
Fixed

Reply:
The way I see this function is that it just waits for the command until it is executed successfully. But we have to stop the command if it takes just to long to succeed. Basically, the main purpose is to wait for success not to wait for timeout. But of course we can change the name.

I think the optional argument ``body`` could come in handy and it doesn't matter if we leave it there.

Reply:
Body can be helpful at some cases, for example if you known that if the command failed, you want to wait some time before trying again because it does not make sense sooner. But we will not use it much probably, since our main use case is to avoid race conditions.

`loop_until_timeout` suggests that we really wait for a timeout to happen, which is not the case. Maybe we can just use `until` which is the bash command? Or `wait_for_condition`, `wait_for_success`?

Reply:
> Body can be helpful at some cases, for example if you known that if the command failed, you want to wait some time before trying again because it does not make sense sooner. But we will not use it much probably, since our main use case is to avoid race conditions.

Okay sure, either way I don't insist.
> 
> `loop_until_timeout` suggests that we really wait for a timeout to happen, which is not the case. Maybe we can just use `until` which is the bash command? Or `wait_for_condition`, `wait_for_success`?

I'm good with `wait_for_condition` or `wait_for_success`

Reply:
Okay, renamed